ruzv01-sp2

vojtar opened the pull request:
Link to application: https://eso.vse.cz/~ruzv01/sp2/

vojtar committed:
ruzv01-sp2

Link to application: https://eso.vse.cz/~ruzv01/sp2/

vojtar committed:
version 1.0

verze aplikace k obhajobě

vojtar committed:
Odstranění style.css 

Ve aktuální verzi je pro stylování použit Bootstrap

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
-const url = "https://api.rawg.io/api/games"
-$.getJSON(url).done((resp) => {
-    console.log(resp)
+const gameForm = $('#input-form');
```
veskery kod dejte do closure `$(document).ready(() => {})`

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+const gameNames = Object.keys(localStorage);
+
+gameNames.forEach((game) => {
+    addGame(game)
```
aktualizace DOMu v cyklu

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+        }
+
+    }).fail((err) => {
+        console.log("hra neexistuje")
```
o tomhle byste mel informovat uzivatele

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+
+        const gameNameContainer = $(`<div class="card-header">${gameName}</div>`);
+
+        const gameImage = $(`<img src="${gameCoverImage}" class="card-img-top" alt="${gameName} cover image" height="250";>`);
```
opravte rozmery obrazku

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+};
+
+const addGame = function(gameNameValue) {
+    $('#game-list').append(loader);
```
co kdyz hru uz mam v seznamu?

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+gameForm.submit((e) => {
+    e.preventDefault();
+    const gameNameValue = gameInput.val().trim().toLowerCase().replace(/\s/g, '-').replace(/'|:/g, '');
+    addGame(gameNameValue)
```
;

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+const addGame = function(gameNameValue) {
+    $('#game-list').append(loader);
+
+    $.getJSON(`${gameDatabaseApi}/${gameNameValue}`).done((resp) => {
```
proc volate stejnou API url dvakrat, (tady v addGame a pak jeste v createGame) ?

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+                <li class="list-group-item"><b>Datum vydání: </b>${gameReleaseDate}</li>
+                <li class="list-group-item"><b>Žánr: </b>${gameGenres}</li>
+                <li class="list-group-item"><b>Metacritic scóre: </b>${gameMetacriticScore}</li>
+                <li class="list-group-item"><b>Platformy: </b>${gamePlatforms}</li>
```
`<strong>`
`gamePlatforms.join(', ')`

nvbach91 commented on www/ruzv01/sp2/main.js:
```diff
@@ -1,4 +1,184 @@
+    const gameNames = Object.keys(localStorage);
+
+    gameNames.forEach((game) => {
+        if (JSON.parse(localStorage.getItem(game)) == 'Odehráno') {
```
lepsi struktura localStorage by byla:
```js
localStorage.games = JSON.stringify({
  "slug1": { "played": true },
  "slug2": { "played": false },
});
```

nvbach91 reviewed (changes requested):
- pridejte tagy k jednotlivym hram
- pridejte moznost vyhledavani her podle search query https://api.rawg.io/docs/#tag/games
- vysledky vyhledavani zobrazte vcetne tlacitek pridat a odebrat (v pripade ze hra uz je v seznamu)

vojtar committed:
Odevzdání opravené verze

- byly přidány tagy ke hrám
- uživatel má možnost vyhledat hru před přidáním
- ze seznamu vyhledaných her může vybrat hru a přidat ji do seznamu
- uživatel nemůže přidat hru, pokud je již v seznamu 
- veškerý kód je zabalen do $(document).ready(() => {})
- addGame a createGame sjednoceno
- aktualizace DOMu v cyklu vyřešena částečně, na konci kódu v main.js je komentář

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
